http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -254,6 +255,34 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       )
     } yield ()
 
+  override def updateDateAccessed(userInfo: UserInfo, workspaceId: WorkspaceId, runtimeName: RuntimeName)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      runtime <- RuntimeServiceDbQueries.getActiveRuntime(workspaceId, runtimeName).transaction
+
+      // If user is creator of the runtime, they should definitely be able to see the runtime.
+      hasPermission <-
+        if (runtime.auditInfo.creator == userInfo.userEmail) F.pure(true)
+        else {
+          checkSamPermission(
+            WsmResourceSamResourceId(WsmControlledResourceId(UUID.fromString(runtime.samResource.resourceId))),
+            userInfo,
+            WsmResourceAction.Write
+          ).map(_._1)
+        }
+
+      _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done auth call for update date accessed runtime permission")))
+      _ <- F
+        .raiseError[Unit](
+          RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied", Some(ctx.traceId))
+        )
+        .whenA(!hasPermission)
+
+      _ <- dateAccessUpdaterQueue.offer(UpdateDateAccessMessage(runtimeName, runtime.cloudContext, ctx.now))
```

Comment:
The original design for this endpoint is for Azure runtimes only, would it make sense to constrain it to that specific cloud context? 
It does not seem that there is anything preventing a user from calling this patch on a gcp runtime, but that could mess up with how autopause works there. What do you think?

Reply:
This may not be applicable to google runtimes, but I also think the direction we should try to head with v2 runtimes is make them as general as possible. The less cloud specific these endpoints the better is my opinion.